counsel.el

```diff
@@ -4564,6 +4564,14 @@ Note: Duplicate elements of `kill-ring' are always deleted."
  '(("d" counsel-yank-pop-action-remove "delete")
    ("r" counsel-yank-pop-action-rotate "rotate")))
 
+(defcustom counsel-yank-delete-selection (when delete-selection-mode 'yank)
+  "Control if `counsel-yank-pop' deletes the active region before yanking.
+This setting rely on the `delete-selection-mode'."
+  :type 'symbolp)
```

Comment:
There's no need for a user option:
- Users can toggle this behaviour at runtime using `delete-selection-mode`.
- Built-in `yank-pop` unconditionally has a `delete-selection` property.